set.mm

```diff
@@ -29066,6 +29066,84 @@ existence hypothesis (to prove the "some" in the conclusion).  Example
       O $.
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+               Intuitionistic logic
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+Intuitionistic (constructive) logic is similar to classical logic
+with the notable omission of ~ ax-3 and theorems such as ~ exmid or
+~ peirce . We mostly treat intuitionistic logic in a separate file, iset.mm,
+which is known as the Intuitionistic Logic Explorer on the web site. However,
+iset.mm has a number of additional axioms (mainly to replace definitions like
+~ df-or and ~ df-ex which are not valid in intitionistic logic) and we want
+to prove those axioms here to demonstrate that adding those axioms in iset.mm
+does not make iset.mm any less consistent than set.mm.
+
+$)
+
+  $( Specialization (intuitionistic logic axiom ax-4).  This is just ~ sp by
+     another name.  (Contributed by Jim Kingdon, 31-Dec-2017.) $)
+  axi4 $p |- ( A. x ph -> ph ) $=
+    ( sp ) ABC $.
+
+  $( Converse of ax-5o (intuitionistic logic axiom ax-i5r).  (Contributed by
+     Jim Kingdon, 31-Dec-2017.) $)
+  axi5r $p |- ( ( A. x ph -> A. x ps ) -> A. x ( A. x ph -> ps ) ) $=
+    ( wal wi hba1 hbim sp imim2i alimi syl ) ACDZBCDZEZNCDLBEZCDLMCAC
+    FBCFGNOCMBLBCHIJK $.
+
+  $( ` x ` is not free in ` A. x ph ` (intuitionistic logic axiom ax-ial).
+     (Contributed by Jim Kingdon, 31-Dec-2017.) $)
+  axial $p |- ( A. x ph -> A. x A. x ph ) $= ( hba1 ) ABC $.
+
+  $( ` x ` is bound in ` E. x ph ` (intuitionistic logic axiom ax-ie1).
+     (Contributed by Jim Kingdon, 31-Dec-2017.) $)
+  axie1 $p |- ( E. x ph -> A. x E. x ph ) $=
+    ( hbe1 ) ABC $.
+
+  $( A key property of existential quantification (intuitionistic logic axiom
+     ax-ie2).  (Contributed by Jim Kingdon, 31-Dec-2017.) $)
+  axie2 $p |- ( A. x ( ps -> A. x ps ) ->
+              ( A. x ( ph -> ps ) <-> ( E. x ph -> ps ) ) ) $=
+    ( wal wi wnf wex wb df-nf 19.23t sylbir ) BBCDECDBCFABECDACGBEHBCIABCJK $.
+
+  $( Axiom of existence (intuitionistic logic axiom ax-i9).  In classical
+     logic, this is equivalent to ~ ax-9 but in intuitionistic logic it needs
+     to be stated using the existential quantifier.  (Contributed by Jim
+     Kingdon, 31-Dec-2017.) $)
+  axi9 $p |- E. x x = y $=
+    ( a9e ) ABC $.
+
+  $( Axiom of Quantifier Substitution (intuitionistic logic axiom ax-10).  This
+     is just ~ ax10 by another name.  (Contributed by Jim Kingdon,
+     31-Dec-2017.) $)
+  axi10 $p |- ( A. x x = y -> A. y y = x ) $=
+    ( ax10 ) ABC $.
```

Comment:
I think this should be `ax-10` not `ax10`.

Reply:
Actually since ax-10 is redundant, we have deprecated its use and have a "New usage is discouraged" tag.   ax10 (which someday will probably be named something more meaningful) is the one to use, being derived from the official axioms.

Norm

Reply:
I meant, of course, classically redundant.  I have no idea if it can be derived from the others intuitionistically, so for iset.mm we still should keep ax-i10 unless and until it can be shown redundant there.

Norm